### PR TITLE
Created UserDaoHibernateImpl, RoleDaoHibernateImpl and BucketDaoHiber…

### PR DESCRIPTION
**Pull request by Korfat**
…nateImpl

### CODE REVIEW

File: src/main/java/internetshop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    private static Logger logger = Logger.getLogger(BucketDaoHibernateImpl.class);
+
+    @Inject
+    private static ItemDao itemDao;
+
+    @Override
+    public Optional<Bucket> create(Bucket bucket) {
+        Long bucketId = null;
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/BucketDaoHibernateImpl.java]: try to use finally with methods create, update and delete in all dao

File: src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("from User where token=:token");
+            query.setParameter("token", token);
+            List<User> list = query.list();
+            return list.stream().findFirst();
+        }
+    }
+
+    @Override
+    public Optional<List<Order>> getOrders(Long userId) {
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java]:
```suggestion
    public List<Order> getOrders(Long userId) {
```

File: src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            logger.error("Failed to add the item into the bucket");
+        }
+        return get(userId);
+    }
+
+    @Override
+    public Optional<User> setUser(Long id, String name, String surname, String login,
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java]: should this method be public?
maybe better remove it from interface?

File: src/main/java/internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 4]
```diff
@@ -8,54 +8,78 @@
...
+            item = session.get(Item.class, id);
+        } catch (Exception e) {
+            logger.error("Can't get user");
         }
+        return Optional.ofNullable(item);
     }
 
     @Override
     public List<Item> getAll() {
+        List<Item> items = null;
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/ItemDaoHibernateImpl.java]: List<Item> items = new ArrayList();

File: src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java [hunk 5]
```diff
@@ -28,45 +27,66 @@
...
+            user = session.get(User.class, id);
+        } catch (Exception e) {
+            logger.error("Can't get user");
         }
+        return Optional.ofNullable(user);
     }
 
     @Override
     public List<User> getAll() {
+        List<User> users = null;
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java]: List<User> users = new ArrayList();

File: src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java [hunk 6]
```diff
@@ -110,41 +137,33 @@ public void delete(Long id) {
     }
 
     @Override
-    public Optional<List<Order>> getOrders(Long userId) {
-        return Optional.empty();
+    public List<Order> getOrders(Long userId) {
+        return null;
```
Comment (Sofasmile) [inline src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java:148]: add implementation

Comment (Korfat) [reply src/main/java/internetshop/dao/hibernate/UserDaoHibernateImpl.java:148]: Тут використовується One-To-Many, який потрібно зробити в наступному ДЗ

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**